exercises/concept/lasagna/.docs/introduction.md

```diff
@@ -3,122 +3,120 @@
+The compiler will infer a suitable type for any expression you use.
 
-Integer literals in Julia are represented in the usual way.
-Underscores may be used as a digit separator.
+## Constants
 
-```julia-repl
-julia> 1
-1
-
-julia> -1234
--1234
+Global variables, created outside any function, are:
+- Allowed.
+- Sometimes necessary.
+- Usually discouraged (only within `*.jl` files; the REPL operates differently).
 
-julia> 1_234_567_890
-1234567890
-```
+If a value needs to be available throughout the program, but is not expected to change, use a constant instead.
 
-### Arithmetic operations
+Prefacing the assignemt with the `const` keyword allows the compiler to generate more efficient code.
 
-The standard prefix and infix operations are available: `+`, `-`, `*`, `%`.
+Accidentally trying to change the `const` value will give a warning:
 
 ```julia-repl
-julia> +16
-16
+julia> const answer = 42
+42
 
-julia> -16
--16
-
-julia> 16 + 6
-22
+julia> answer = 24
+WARNING: redefinition of constant Main.answer. This may fail, cause incorrect answers, or produce other errors.
+24
+```
 
-julia> 16 - 6
-10
+## Arithmetic operators
 
-julia> 16 * 6
-96
+These mostly work conventionally:
 
-julia> 16 % 6
-4
+```julia
+2 + 3  # 5 (addition)
+2 - 3  # -1 (subtraction)
+2 * 3  # 6 (mutlplication)
+8 / 2  # 4.0 (division with floating-point result)
+8 % 3  # 2 (remainder)
 ```
 
-#### Division
+## Functions
 
-Dividing two numbers with the `/` operator will result in a floating point value.
-To perform integer division
+There are two common ways to define a named function in Julia:
 
-- use the `div(x, y)` function, or
-- use the `÷` operator (Julia source code is unicode-aware)
+1. Using the `function` keyword
 
-```julia-repl
-julia> 16 / 6
-2.6666666666666665
+    ```julia
+    function muladd(x, y, z)
+        x * y + z
+    end
+    ```
 
-julia> div(16, 6)
-2
+    Indentation by 4 spaces is conventional for readability, but the compiler ignores this.
+    The `end` keyword is essential: more like Ruby than like Python.
 
-julia> 16 ÷ 6
-2
-```
+    Note that we could have written `return x * y + z`. 
+    However, Julia functions always return the last expression evaluated, so the `return` keyword is optional.
+    Many programmers prefer to include it to make their intentions more explicit.
 
-~~~~exercism/note
-It's natural to use Unicode symbols in Julia source files, typically in mathematical expressions.
-When using the Julia REPL, or in other Julia editing environments, the division symbol can be entered by typing `\div` followed by the `Tab` key.
-More details can be found in the manual at [Unicode Input][unicode].
+2. Using the "assignment form"
 
-[unicode]: https://docs.julialang.org/en/v1/manual/unicode-input/#Unicode-Input
-~~~~
+    ```julia
+    muladd(x, y, z) = x * y + z
+    ```
 
-## Comments
+    This is most commonly used for one-line function definitions or mathematical functions, where the function body is a single expression.
```

Comment:
What's the difference between one-line function definitions and mathematical functions? We don't give context for what a mathematical function is. Perhaps we can say this is for making concise single-expression functions and that'd be clear enough.

Reply:
Agree

Reply:
Switched the latest commit to your shorted wording.